diff --git a/src/Service/MenuLinkTreeMenuToken.php b/src/Service/MenuLinkTreeMenuToken.php
index 19727b7..2e49227 100644
--- a/src/Service/MenuLinkTreeMenuToken.php
+++ b/src/Service/MenuLinkTreeMenuToken.php
@@ -74,6 +74,7 @@ class MenuLinkTreeMenuToken extends MenuLinkTree {
    */
   protected function buildItems(array $tree, CacheableMetadata &$tree_access_cacheability, CacheableMetadata &$tree_link_cacheability) {
     $items = [];
+	$current_path = \Drupal::request()->getRequestUri();
 
     foreach ($tree as $data) {
       /** @var \Drupal\Core\Menu\MenuLinkInterface $link */
@@ -142,6 +143,10 @@ class MenuLinkTreeMenuToken extends MenuLinkTree {
       if (isset($data->options)) {
         $element['url']->setOptions(NestedArray::mergeDeep($element['url']->getOptions(), $data->options));
       }
+	  if ($element['url']->toString() == $current_path) {
+        $element['url']->setOption('attributes', array('class' => array('is-active')));
+      }
+
       $element['original_link'] = $link;
       $items[$link->getPluginId()] = $element;
     }
